app/main.py

```diff
@@ -1,13 +1,42 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            } for error in errors
+        ],
+        "path": file_path,
+        "status": "failed" if len(errors) != 0 else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {
+                    "line": info["line_number"],
+                    "column": info["column_number"],
+                    "message": info["text"],
+                    "name": info["code"],
+                    "source": "flake8",
+                } for info in value
+            ],
+            "path": key,
+            "status": "failed" if len(value) != 0 else "passed"
+        } for key, value in linter_report.items()
+    ]
```

Comment:
Do not use code duplication here, evoke the `format_single_linter_file` function to perform this operation.